docs/manual/config/without-dm.rst

```diff
@@ -0,0 +1,64 @@
+====================
+Running from systemd
+====================
+
+This case will cover automatic login to Qtile after booting the system without
+using display manager. It logins in virtual console and init X by running
+through session.
+
+Automatic login to virtual console
+----------------------------------
+
+To get login into virtual console as an example edit `getty` service by running
+`systemctl edit getty@tty1` and add instructions:
+
+::
+    :caption: /etc/systemd/system/getty@tty1.service.d/override.conf
+
+    [Service]
+    ExecStart=
+    ExecStart=-/usr/bin/agetty --autologin username --noclear %I $TERM
+
+`username` should be changed to current user name.
+
+Check more for other `examples <https://wiki.archlinux.org/index.php/Getty#Automatic_login_to_virtual_console>`_.
+
+Qtile session
+-------------
+
+Starting Qtile directly will not preserve running applications on Qtile
+restarts by `qtile-cmd -o cmd -f restart` or other ways in Qtile itself. This
+can be solved by starting simple script loop.
+
+Autostart X session
+-------------------
+
+After login X session should be started. That can be done by `.bash_profile` if
+bash is used or `.zprofile` in case of zsh. Other shells can be adjusted by
+given examples.
+
+.. code-block:: bash
+
+    if systemctl -q is-active graphical.target && [[ ! $DISPLAY && $XDG_VTNR -eq 1 ]]; then
+      exec startx
+    fi
+
+And to start Qtile itself `.xinitrc` should be fixed:
+
+::
+
+    # some apps that should be started before Qtile, ex.
+    #
+    #   [[ -f ~/.Xresources ]] && xrdb -merge ~/.Xresources
+    #   ~/.fehbg &
+    #   nm-applet &
+    #   blueman-applet &
+    #   dunst &
+    #
+    # or
+    #
+    #   source ~/.xsession
+
+    while true; do
+        exec qtile
```

Comment:
`exec` exits from the script when it is done, so the loop won't work. But do you even need a loop? Why exit Qtile just to run it again, when you can just restart it? The only reason you'd exit Qtile on a DM-less setup is if you want the TTY, no?

Reply:
you are right about `exec`. Romeved and it worked. tych0 right about purpose of this loop